pkg/nodes/enricher/enricher_impl.go

```diff
@@ -126,20 +126,17 @@ func (e *enricherImpl) enrichNodeWithScanner(node *storage.Node, nodeInventory *
 	var scan *storage.NodeScan
 	var err error
 
-	var scannerVersion int
-
 	scanStartTime := time.Now()
 	if scanner.Type() == types.ScannerV4 {
 		scan, err = scanner.GetNodeInventoryScan(node, nil, indexReport)
-		scannerVersion = 4
+		e.metrics.SetScanDurationTime(scanStartTime, scanner.Name(), err)
+		e.metrics.SetNodeInventoryNumberComponents(len(indexReport.GetContents().GetPackages()), node.GetClusterName(), node.GetName(), scanner.Name())
 	} else {
 		scan, err = scanner.GetNodeInventoryScan(node, nodeInventory, nil)
-		scannerVersion = 2
+		e.metrics.SetScanDurationTime(scanStartTime, scanner.Name(), err)
+		e.metrics.SetNodeInventoryNumberComponents(len(nodeInventory.GetComponents().GetRhelComponents()), node.GetClusterName(), node.GetName(), scanner.Name())
```

Comment:
Could you please check that `scanner.Name()` returns values `2` or `4` as we expect them in the metric?
I see that the name being returned in that function depends on the integrations and I am not sure how they are named.

Reply:
I've unified the scanner version strings across all metrics